packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -30,19 +31,26 @@ function BlockListAppender( {
 	rootClientId,
 	canInsertDefaultBlock,
 	isLocked,
-	renderAppender: CustomAppender,
+	renderAppender: CustomAppender, // Deprecated.
+	appender,
 	className,
 	selectedBlockClientId,
 	tagName: TagName = 'div',
 } ) {
-	if ( isLocked || CustomAppender === false ) {
+	if ( CustomAppender ) {
+		deprecated( 'renderAppender prop of the BlockListAppender component', {
+			since: '5.7',
```

Comment:
Is it correct to use the `5.7` here?

Reply:
Also, I added the deprecation only in the component that implemented the changes, instead of all the components that receive this prop to avoid multiple warning deprecations.

Reply:
I think it should be the plugin version rather than the WordPress version. Best to go for the next one (10.8) and then if it misses it can be updated.

Reply:
My previous comment about the deprecation message applies here to:
> I think most implementers will specify this prop on the InnerBlocks component or useInnerBlocksProps hook. They probably won't be aware of BlockListAppender.

> It might be easiest to omit the component name from the message, or change the wording to 'renderAppender prop of InnerBlocks component or useInnerBlocksProps hook'.

Reply:
https://github.com/WordPress/gutenberg/commit/67c2525c25dceef43900aeaf8fedf8b2a4d758a0